Applications/Utils/SWMMConverter/SWMMInterface.h

```diff
@@ -0,0 +1,136 @@
+/**
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ */
+
+#ifndef SWMMINTERFACE_H_
+#define SWMMINTERFACE_H_
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <logog/include/logog.hpp>
+
+namespace GeoLib {
+    class GEOObjects;
+    class Point;
+    class Polyline;
+}
+
+namespace MeshLib {
+    class Mesh;
+    class Node;
+    class Element;
+}
+
+enum class SwmmObject
+{
+    NODE,
+    LINK,
+    SUBCATCHMENT,
+    SYSTEM
+};
+
+/**
+ * Interface for reading files used within the
+ * Storm Water Management Model (SWMM) and converting
+ * those into corresponding OGS data structures.
+ */
+class SwmmInterface
+{
+public:
+    SwmmInterface(std::string const& swmm_base_name);
+
+    SwmmInterface(MeshLib::Mesh* mesh, std::string const& swmm_base_name);
+
+    /// Reading a SWMM input file and conversion into OGS line mesh.
+    MeshLib::Mesh* SwmmInputToLineMesh();
+
+    /// Checking if file is a SWMM input file
+    static int isSwmmInputFile(std::string const& inp_file_name);
+    
+    /// Checking if file is a SWMM output file
+    int isSwmmOutputFile(std::string const& inp_file_name);
+
+    int addResultsToMesh(MeshLib::Mesh &mesh, std::vector<double> data);
+
+    /// Reading a SWMM input file and conversion into OGS geometry.
+    static int SwmmInputToGeometry(std::string const& inp_file_name, GeoLib::GEOObjects &geo_objects);
+
+
+private:
+    /// Checking if file is a SWMM input file
+    static int isSwmmFile(std::string const& inp_file_name, std::ifstream &in, std::string const& ext);
+
+    /// Reading points from SWMM input file and converting them into OGS point vector.
+    template <typename T>
+    static int readCoordinates(std::ifstream &in, std::vector<T*> &points, std::vector<std::string> &names)
```

Comment:
I'd rather keep it int for all functions so I can give meaningful error messages from gui.